### PR TITLE
	added STAR alignment, tagging exons and genes, and collapsing on barcodes based on known used barcodes

### PR DESCRIPTION
**Pull request by Gerbenvandervries**

### CODE REVIEW

File: generate_template.sh [hunk 1]
```diff
@@ -57,4 +57,4 @@ mainParameters=${WORKDIR}/generatedscripts/${PROJECT}/parameters.csv;\
 worksheet=${WORKDIR}/generatedscripts/${PROJECT}/${PROJECT}.csv;\
 parameters_species=${WORKDIR}/generatedscripts/${PROJECT}/parameters.${SPECIES}.csv;\
 parameters_environment=${WORKDIR}/generatedscripts/${PROJECT}/parameters.${ENVIRONMENT}.csv;\
-ngsversion='NGS_ScRNA/1.0.1-Molgenis-Compute-v16.08.1-Java-1.8.0_74';"
+ngsversion='NGS_ScRNA/1.1.1-Molgenis-Compute-v16.08.1-Java-1.8.0_74';"
```
Comment (RoanKanninga) [inline generate_template.sh]: You can better do it like this: ngsversion=$(module list | grep -o -P 'NGS_ScRNA(.+)');
Then you only have to change it once

File: parameters.homo_sapiens.csv [hunk 2]
```diff
@@ -9,6 +9,7 @@ indicesDir,${dataDir}/ftp.broadinstitute.org/bundle/2.8/${genome}/
 dbSNPDir,${dataDir}/dbSNP/
 hisatIndex,${dataDir}/ftp.broadinstitute.org/bundle/2.8/${genome}/${hisatVersion}/${indexFolderName}
 kallistoIndex,${dataDir}/ftp.ensembl.org/pub/release-${ensembleReleaseVersion}/fasta/homo_sapiens/Homo_sapiens.GRCh37.${ensembleReleaseVersion}.cdna.all.fa.idx
+StarIndex,${dataDir}/ftp.broadinstitute.org/bundle/2.8/${genome}/${StarVersion}/
```
Comment (RoanKanninga) [inline parameters.homo_sapiens.csv:12]: you are using capital as first character?

File: protocols/Kallisto.sh [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+makeTmpDir ${intermediateDir}
+tmpIntermediateDir=${MC_tmpFile}
+
+echo "## "$(date)" Start $0"
+echo "ID (project-internalSampleID-lane): ${project}-${externalSampleID}-L${lane}"
+
+mkdir -p ${intermediateDir}/Kallisto/
+
+uniqueID="${project}-${externalSampleID}-L${lane}"
+
```
Comment (RoanKanninga) [inline protocols/Kallisto.sh:32]: No need for indentation, right?

File: protocols/STAR.sh [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+module list
+
+makeTmpDir ${intermediateDir}
+tmpIntermediateDir=${MC_tmpFile}
+
+echo "## "$(date)" Start $0"
+echo "ID (project-internalSampleID-lane): ${project}-${externalSampleID}-L${lane}"
+
+uniqueID="${project}-${externalSampleID}-L${lane}"
+
```
Comment (RoanKanninga) [inline protocols/STAR.sh:27]: Indentation is not needed

File: generate_template.sh (hunk 1 above)
Comment (Gerbenvandervries) [reply generate_template.sh]: done

**Pull request merged**